addons/mrp_account/tests/test_bom_price.py

```diff
@@ -254,3 +254,85 @@ def test_02_compute_byproduct_price(self):
         self.assertEqual(self.dining_table.standard_price, 137.5, "After computing price from BoM price should be 137.5")
         scrap_wood.button_bom_cost()
         self.assertEqual(scrap_wood.standard_price, 20.63, "After computing price from BoM price should be 20.63")
+
+
+    def test_compute_average_price_with_different_company(self):
+        """Ensure _compute_average_price correctly fetches price from the intended company"""
+
+        # Create Companies
+        self.company_a = self.env['res.company'].create({'name': 'Company A'})
+        self.company_b = self.env['res.company'].create({'name': 'Company B'})
+
+        # Create Product Category with cost method other than 'standard'
+        self.product_category = self.env['product.category'].with_company(self.company_b).create({
+            'name': 'Test Category',
+            'property_cost_method': 'average',
+        })
+
+        # Product for Company B
+        self.main_product = self.env['product.product'].create({
+            'name': 'Main Product',
+            'type': 'product',
+            'standard_price': 101.11,
+            'company_id': self.company_b.id,
+            'categ_id': self.product_category.id,
+        })
+
+        # Create additional product for the BOM
+        self.product_b = self.env['product.product'].with_company(self.company_b).create({
+            'name': 'Product B',
+            'type': 'product',
+            'standard_price': 100,
+            'company_id': self.company_b.id,
+            'categ_id': self.product_category.id,
+        })
+
+        # Create Bill of Materials (BoM) for the main product
+        self.bom = self.env['mrp.bom'].with_company(self.company_b).create({
+            'product_id': self.main_product.id,
+            'product_tmpl_id': self.main_product.product_tmpl_id.id,
+            'product_qty': 1.0,
+            'type': 'phantom',
+            'bom_line_ids': [(0, 0, {'product_id': self.product_b.id, 'product_qty': 1})],
+        })
+
+        # Create stock moves for Product B
+        self.stock_moves = self.env['stock.move'].create([
```

Comment:
look at another multicompany test (e.g. `test_so_and_multicompany`) to see how to properly set up the move values